handler.js

```diff
@@ -0,0 +1,125 @@
+'use strict';
+
+const aws = require('aws-sdk')
+const s3 = new aws.S3({ apiVersion: '2006-03-01' });
+const fs = require('fs');
+const path = require("path");
+const saf = require('@mitre/saf');
+const {createWinstonLogger} = require("./lib/logger.js");
+
+async function getObject(bucket, objectKey) {
+    try {
+        const params = {
+            Bucket: bucket,
+            Key: objectKey
+        }
+
+        const data = await s3.getObject(params).promise();
+
+        return data.Body.toString('utf-8');
+    } catch (e) {
+        throw new Error(`Could not retrieve file from S3: ${e.message}`)
+    }
+}
+
+const uploadFile = (fileName, bucket, key) => {
+    const fileContent = fs.readFileSync(fileName);
+
+    const params = {
+        Bucket: bucket,
+        Key: key, // File name you want to save as in S3
+        Body: fileContent
+    };
+
+    s3.upload(params, function(err, data) {
+        if (err) {
+            throw err;
+        }
+        console.log(`File uploaded successfully. ${data.Location}`);
+    });
+};
+
+const getInputFileName = (key) => {
+    return path.basename(key);
+};
+
+const getOutputFileName = (input_file_name, configData) => {
+    const input_file_ext = path.extname(input_file_name);
+    const file_name_no_ext = path.basename(input_file_name, input_file_ext);
+    return file_name_no_ext + configData['output-extension'];
+};
+
+const getConfigData = () => {
+    const config = {
+        "input-bucket": process.env.INPUT_BUCKET,
+        "input-prefix": process.env.INPUT_PREFIX || "",
+        "output-bucket": process.env.OUTPUT_BUCKET || process.env.INPUT_BUCKET,
+        "output-prefix": process.env.OUTPUT_PREFIX || "results/",
+        "output-extension": process.env.OUTPUT_EXTENSION || "_results.json",
+        "output-enabled": process.env.OUTPUT_ENABLED || true,
+    }
+
+    return config;
+};
+
+async function runSaf(command_string) {
+    if (!command_string) {
+        throw new Error("SAF CLI Command String argument is required.");
+    }
+    
+    const saf_command = command_string.split(' ');
+    
+    const allowable_topics = ['convert', 'generate', 'harden', 'scan', 'validate', 'view'];
+    const topic = saf_command[0].split(':')[0];
+    
+    if (!allowable_topics.includes(topic)) {
+        throw new Error("The command string did not include one of the allowable topics: " + allowable_topics.join(', ') + ". Please reference the documentation for more details.");
+    }
+    
+    const command = saf_command[0].split(':')[1];
+    
+    if (topic == "view" & command == "heimdall") {
```

Comment:
since we currently don't have anything for 'scan' or 'harden' either, should we spit out a warning here?

Reply:
the warning/message that those functions spit out themselves might be fine

Reply:
The responses from SAF CLI will give the user the latest information as the topics and commands adapt. So more restrictive filtering can be omitted here and left to SAF CLI